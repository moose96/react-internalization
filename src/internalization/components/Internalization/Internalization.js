import React from 'react';
import PropTypes from 'prop-types';

import InternalizationContext from '../../context';
import checkTranslations from '../../utils/CheckTranslations';

function Internalization({ data, lang, defaultLang, children }) {
  let translations = {};

  const checkedTranslations = checkTranslations(data);
  if (!checkedTranslations.state) {
    throw Error(checkedTranslations.data);
  }

  if (!data[lang]) {
    translations = data[defaultLang];
  } else {
    translations = data[lang];
  }

  return (
    <InternalizationContext.Provider value={translations}>
      {children}
    </InternalizationContext.Provider>
  )
}

Internalization.propTypes = {
  data: PropTypes.objectOf(PropTypes.object).isRequired,
  lang: PropTypes.string,
  defaultLang: PropTypes.string.isRequired,
  children: PropTypes.node
};


export default Internalization;