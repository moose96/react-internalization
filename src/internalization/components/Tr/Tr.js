import React from 'react';
import PropTypes from 'prop-types';

import useTranslation from '../../hooks/useTranslation';

function Tr({ component, iKey, ...props }) {
  const text = useTranslation(iKey);

  if (text) {
    return React.cloneElement(component, {...props}, text);
  } else {
    return component;
  }
}

Tr.propTypes = {
  component: PropTypes.node.isRequired,
  iKey: PropTypes.string.isRequired
}

export default Tr;

//
//<Tr component={<h2 />} name={'MessageBox.Yes'} />
//<Tr component={<a href="" />} name={'About'} />
//<Tr name={'MessageBox.Yes}>
//  <h2></h2>
//</Tr>
//const keys = useTranslations('MessageBox');