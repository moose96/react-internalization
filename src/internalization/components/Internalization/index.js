import Internalization from './Internalization';

export default Internalization;