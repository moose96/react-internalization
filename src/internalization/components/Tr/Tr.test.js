import React from 'react';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';

import Internalization from '../Internalization';
import Tr from './Tr';

const translations = {
  'en': {
    'test': 'hello'
  }
}

describe('test Tr consumer', () => {
  it('should render given component', () => {
    const components = (
      <Internalization data={translations} lang="en">
        <Tr component={<h1>a</h1>} iKey='test' />
      </Internalization>
    );

    const snapshot = TestRenderer
      .create(components)
      .toJSON();

    expect(snapshot).toMatchSnapshot();

    const wrapper = shallow(components);
    expect(wrapper.html()).toEqual('<h1>hello</h1>');
  });

  it('should render default component', () => {
    const wrapper = shallow(
      <Internalization data={translations} lang="en">
        <Tr component={<p>default</p>} iKey="notexist" />
      </Internalization>
    );

    expect(wrapper.html()).toEqual('<p>default</p>');
  })

  it('should render component with props', () => {
    const wrapper = shallow(
      <Internalization data={translations} lang="en">
        <Tr component={<a>default</a>} iKey="test" href="/test" />
      </Internalization>
    );

    expect(wrapper.html()).toEqual('<a href="/test">hello</a>');
  })
});
