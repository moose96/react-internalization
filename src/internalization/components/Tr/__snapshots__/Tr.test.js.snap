// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tr consumer should render given component 1`] = `
<h1>
  hello
</h1>
`;
